src/battle.ts

```diff
@@ -306,25 +306,25 @@ class Pokemon implements PokemonDetails, PokemonHealth {
 		if (!this.hasMovestatus(volatile)) return;
 		delete this.movestatuses[volatile];
 	}
-	addMovestatus(volatile: ID) {
+	addMovestatus(volatile: ID, ...args: any[]) {
 		volatile = toID(volatile);
 		if (this.hasMovestatus(volatile)) return;
-		this.movestatuses[volatile] = [volatile];
+		this.movestatuses[volatile] = [volatile, ...args] as EffectState;
 		this.side.battle.scene.addEffect(this, volatile);
 	}
 	hasMovestatus(volatile: ID) {
 		return !!this.movestatuses[volatile];
 	}
-	clearMovestatuses() {
+	clearMovestatusesExcept(exceptions: ID[] = []) {
```

Comment:
I wouldn't rename this; `clearMovestatuses(except: ID[] = []) {` is clear enough.

Reply:
I also don't think this needs to be an array.

On second thought, I would probably do this as `clearMovestatuses(move: Move)`, and then check for Fury Cutter inside the function.

Reply:
I chose array because I'm implementing Metronome (item) much the same way in https://github.com/smogon/pokemon-showdown-client/pull/1792. Would I be able to keep the array for that purpose?